fuzz/roundtrip.cpp

```diff
@@ -219,26 +220,31 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
     /**
      * Transcoding from UTF-16LE to UTF-8.
      */
+    // We copy to avoid alignment issues.
+    std::unique_ptr<char16_t[]> utf16_source{new char16_t[source.size() / 2]};
+    if (source.data() != nullptr) {
+      std::memcpy(utf16_source.get(), source.data(), source.size() / 2 * 2);
+    }
 
     // Get new source data here as this will allow the fuzzer to optimize it's
     // input for UTF16-LE.
     source = fdp.ConsumeRandomLengthString(kMaxStringSize);
     bool validutf16le =
-        e->validate_utf16le((char16_t*)source.c_str(), source.size() / 2);
-    auto rutf16le = e->validate_utf16le_with_errors((char16_t*)source.c_str(),
-                                                    source.size() / 2);
+        e->validate_utf16le(utf16_source.get(), source.size() / 2);
```

Comment:
this uses the data from the copy made on line 223, but the intention (as seen on line 231) was to use other data.

maybe turn rows 223-227 into a function which returns a unique_ptr with a copy of the data?

Reply:
Fixed. I am now scoping the copies.